### PR TITLE
Add $USER to all Docker image names.

### PR DESCRIPTION
**Pull request by egrimley-arm**
This is another attempt to do what #650 was probably trying to do.

Implemented on top of #656, which is expected to be merged first, so "Files changed" shows more than one PR. It probably makes more sense to look at the top commit.

### CODE REVIEW

**Comment by mathias-arm** [comment]
I don't think we need the intermediate images to be renamed only the "leaf" one. I am worried also about the consequences on the `ci` image build.

**Comment by egrimley-arm** [comment]
> I don't think we need the intermediate images to be renamed only the "leaf" one.

That might be what we already have (after #654 is applied).

> I am worried also about the consequences on the `ci` image build.

I'm worried about that, too, and generally how there a lot of things to test here.

File: third-party/rust-mbedtls
Comment (mathias-arm) [inline third-party/rust-mbedtls:1]: [9232612](https://github.com/veracruz-project/veracruz/pull/657/commits/92326124a717e700a3f1539b89c0aa927adaf813) includes this change which is surprising given the commit name. If I understand it changes from https://github.com/veracruz-project/rust-mbedtls/commit/7deda7121d54e97693c1e800bb717add116b80ef to https://github.com/veracruz-project/rust-mbedtls/commit/843291002b226a38821acda887bc05548dec2f2f which is to me a regression.

**Review by mathias-arm — CHANGES_REQUESTED** [review]

File: third-party/rust-mbedtls
Comment (egrimley-arm) [reply third-party/rust-mbedtls:1]: Oh, well spotted. It wouldn't be the first time that I've accidentally included a change to a submodule by changing to another branch, not doing `git submodule update`, and then using `git commit -a`.

**Comment by mathias-arm** [comment]
Closing in favor of #656 which I think is the best solution.

**Pull request closed without merging**